sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ErrorHandling.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.javasdk.impl
+
+import org.slf4j.MDC
+
+import java.util.UUID
+
+/**
+ * INTERNAL API
+ */
+private[javasdk] object ErrorHandling {
+
+  def withCorrelationId[T](block: String => T): T = {
+    val correlationId = UUID.randomUUID().toString
+    MDC.put("correlationID", correlationId)
+    try {
+      block(correlationId)
+    } finally {
+      MDC.remove("correlationID")
+    }
+  }
```

Comment:
Turns out you don't really see mdc values in the console as I was hoping. I wonder if it is worth doing like this and see if we can get that added, or just put the id in the log entry message string which will work right away?

Reply:
but MDC values will show up in production logs (otherwise we have a problem) and isn't it for that this is most useful?

Reply:
on the other hand, if we don't log it as mdc on proxy side we could just make it the same here, i.e. include in log message instead

(that also might make it easier for having more similar logging from JS)

Reply:
I wonder if I could use MDC but use the layout to include in the log message as well for now (until log viewer/production logs can show it)

Reply:
Verified that the MDC value is visible through the akkasls logs command, but with the correlation in the log entry through the formatter you also see it already through the console log ui (otoh you only see the first line of the log message so it's anyway not super useful there)